interfaces/builtin/raw_volume.go

```diff
@@ -0,0 +1,167 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2019 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+import (
+	"fmt"
+	"path/filepath"
+	"regexp"
+	"strings"
+
+	"github.com/snapcore/snapd/interfaces"
+	"github.com/snapcore/snapd/interfaces/apparmor"
+	"github.com/snapcore/snapd/interfaces/udev"
+	"github.com/snapcore/snapd/snap"
+)
+
+const rawVolumeSummary = `allows read/write access to specific disk partition`
+
+const rawVolumeBaseDeclarationPlugs = `
+  raw-volume:
+    allow-installation: false
+    deny-auto-connection: true
+`
+
+const rawVolumeBaseDeclarationSlots = `
+  raw-volume:
+    allow-installation:
+      slot-snap-type:
+        - core
+        - gadget
+    deny-auto-connection: true
+`
+
+// Only allow disk device partitions; not loop, ram, CDROM, generic SCSI,
+// network, tape, raid, etc devices
+const rawVolumeConnectedPlugAppArmorPath = `
+# Description: can access disk partition read/write
+%s rw,
+
+# needed for write access
+capability sys_admin,
+
+# allow read access to sysfs and udev for block devices
+@{PROC}/devices r,
+/run/udev/data/b[0-9]*:[0-9]* r,
+/sys/block/ r,
+/sys/devices/**/block/** r,
+`
+
+// The type for this interface
+type rawVolumeInterface struct{}
+
+// Getter for the name of this interface
+func (iface *rawVolumeInterface) Name() string {
+	return "raw-volume"
+}
+
+func (iface *rawVolumeInterface) StaticInfo() interfaces.StaticInfo {
+	return interfaces.StaticInfo{
+		Summary:              rawVolumeSummary,
+		BaseDeclarationPlugs: rawVolumeBaseDeclarationPlugs,
+		BaseDeclarationSlots: rawVolumeBaseDeclarationSlots,
+		ImplicitOnCore:       true,
```

Comment:
I suspect now that this carries paths being implicit doesn't make sense, am I missing something?

If meaningful we can add code hotplug-like code at a later point to list device into slots on classic though.

Reply:
I was thinking about the future potential of hotplug code, yes, but since it isn't there now, probably best to remove it.

Reply:
Done